base_layer/core/src/blocks/blockheader.rs

```diff
@@ -68,7 +68,7 @@ pub struct BlockHeader {
     #[serde(with = "hash_serializer")]
     pub prev_hash: BlockHash,
     /// Timestamp at which the block was built.
-    pub timestamp: DateTime<Utc>,
+    pub timestamp: u64,
```

Comment:
The timestamp functions all take or return i64, so shouldn't this be i64?
```Rust
fn from_timestamp(secs: i64, nsecs: u32) -> NaiveDateTime
fn timestamp(&self) -> i64
```

Reply:
Chrono's timestamps do i64. 
We dont want a negative timestamp. So its safer to go u64.

Reply:
prost's Timestamp is also `i64`, so we'd be cutting off precision anyway everytime we hit the wire.

Maybe we should just use `i64` for maximum compatibility.

Reply:
I assume timestamps < 0 mean before 1970

Reply:
Might be, but we dont care about timestamps before 2019. 
My concern is that u64's are safer for for calculations, as we dont want negatives there. I would prefer making a clean seperation where we do convert. Converting everythime we use it might becomes messy, and then we just might as well stick to datetime.